flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/FileMergingSnapshotManagerBase.java

```diff
@@ -278,7 +300,11 @@ public SegmentFileStateHandle closeStreamAndCreateStateHandle(
                             returnPhysicalFileForNextReuse(subtaskKey, checkpointId, physicalFile);
 
                             return new SegmentFileStateHandle(
-                                    physicalFile.getFilePath(), startPos, stateSize, scope);
+                                    getManagedDirStateHandle(subtaskKey, scope),
```

Comment:
Is it possible for some cp happened to be with no state, and the directory is deleted by JM?